install.sh

```diff
@@ -1,20 +1,6 @@
 #!/usr/bin/env bash
-function doInstall {
-	if ! [ -L ~/bin ] && ! [ -d ~/bin ]; then
-		echo make dir
-		mkdir ~/bin
-	else
-		echo dir '~/bin' exist
-	fi
-	cd /tmp && \
-	git clone https://github.com/imachug/win-sudo.git && \
-	mv win-sudo/su* ~/bin && \
-	rm -rf win-sudo
-}
-
-function doFail {
-	echo "failed to install, sorry :'(";
-	exit 1
-}
-doInstall || doFail
-echo "enjoy :)"
+echo "Downloading win-sudo...";trap 'echo "Failed to install, sorry :(";exit 1' ERR
+mkdir -p ~/bin/win-sudo && cd "$_";git init -q && git config core.sparsecheckout true;echo s/ >> .git/info/sparse-checkout
+git remote add -mf origin https://github.com/DemonixCraft/win-sudo.git && git pull -q origin master
+echo "source ~/bin/win-sudo/s/path.sh" | tee -a ~/.bashrc >/dev/null 2>&1 && echo "Win-sudo successfully installed!"
+rm -rf ~/.bash_profile >/dev/null 2>&1;exec bash
```

Comment:
Not sure about this thing. Probably it would be better if you added `source ~/bin/win-sudo/s/path.sh` line to `bash_profile` as well, because some people may have some scripts here instead of `bashrc`.

Reply:
tee appends the line to ~/.bashrc

Reply:
I'll look at the options available and see if its possible.

Reply:
~/.bash_profile is usually found in the $root_dir of linux distros. It's generally sourced when your logging in with SSH. It's much more reliable to put the PATH variable in ~/.bashrc because it is sourced upon startup. Putting it in ~/.bash_profile as well means you can activate sudo with one of two ways per shell spawn : 
* Login with SSH
* Run a sourcing command for ~/.bashrc or ~/.bash_profile

My goal here is to make win-sudo completely auto-deployable and only require typing one line of code for complete and quick setup.
 I could either prompt the user to confirm the removal of ~/.bash_profile or temporarily move ~/.bash_profile to a sub dir where it won't conflict with sudo.